Model/Adapter/AmazonPayAdapter.php

```diff
@@ -128,7 +128,7 @@ protected function createPrice($amount, $currencyCode)
                 $amount = round($amount);
                 break;
             default:
-                $amount = (float) $amount;
+                $amount = (float)number_format((float)$amount, 2);
```

Comment:
```suggestion
                $amount = (float)number_format($amount, 2, '.', '');
```

Will get thousands separators with `number_format` unless we specify not to use them:
`php > echo number_format(12345.67890,2);`
`12,345.68`